### ISSUE TITLE
fake data --remote doesnt work with the current web ui

### ISSUE THREAD

**Issue by Akromx16** [rib/gputop#105]
Travis at the moment checks 2 builds out of 6 with the --fake option, which runs the ncurses UI. However, when used with --remote as well, the website gets into all kinds of assertion and doesn't display anything anything useful

**Comment by rib** [comment]
B.t.w I think travis has some documentation specifically for projects that need to be able to run web browser based tests, so it might not be super tricky to be able to smoke test the web ui as part of travis builds.

Just skimming quickly, this might help us hook up a smoke test: https://docs.travis-ci.com/user/gui-and-headless-browsers/

**Comment by rib** [comment]
Looking at this a bit more, I think to start with just using Xvfb + running Firefox (which is apparently installed on all Travis CI instances) could be fairly straight forward. Xvfb is just an X server that doesn't output to a real display and can be launched before running a test in our .travis.yml file with something like:
```
- "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16"
```
Then we'd just need to set the DISPLAY environment variable = `:99` before running firefox:
```- "export DISPLAY=:99.0"```

We can run gputop in the background, redirected into a log file something like:
```- gputop --remote --fake [options...] >/tmp/gputop-remote.log &```

And launch the web ui something like:
```- firefox http://localhost:7890```

I think something that could help a lot with this kind of testing though is if we could add some protocol for the web-ui to send 'TEST: <message>' log messages back to the gputop server which would be written to the /tmp/gputop-remote.log file. That way we can place test log messages in key parts of the UI that we can verify have been successfully executed if we see a corresponding 'TEST: foo' message in the /tmp/gputop-remote.log file.

**Comment by rib** [comment]
Even without any extra test log message protocol, there are already enough messages that the gputop server will output after the web ui successfully connects that we could grep for a few key messages that could indicate successfully connecting and receiving some data.

**Comment by rib** [comment]
resolved recently in #115, thanks
